gridcapa-swe-commons/src/main/java/com/farao_community/farao/gridcapa_swe_commons/shift/SweNetworkShifter.java

```diff
@@ -151,29 +148,46 @@ private boolean isShiftSucceed(double mismatchEsPt,  double mismatchEsFr) {
             String zoneId = entry.getKey();
             double asked = entry.getValue();
             String logApplyingVariationOnZone = String.format("[%s] : Applying variation on zone %s (target: %.2f)", direction, zoneId, asked);
             LOGGER.info(logApplyingVariationOnZone);
             double done = zonalScalable.getData(zoneId).scale(network, asked, scalingParameters);
             if (Math.abs(done - asked) > DEFAULT_SHIFT_EPSILON) {
-                String logWarnIncompleteVariation = String.format("[%s] : Incomplete variation on zone %s (target: %.2f, done: %.2f)",
+                String logWarnIncompleteVariation = String.format("[%s] : Incomplete shift on zone %s (target: %.2f, done: %.2f)",
                         direction, zoneId, asked, done);
                 LOGGER.warn(logWarnIncompleteVariation);
-                limitingCountries.add(zoneId);
+                incompleteShiftCountries.put(zoneId, done - asked);
             }
         }
         // During the shift some generators linked to the main network with a transformers are not connected correctly
         // Waiting for a fix in powsybl-core, we connect the transformers linked to these generators to be correctly connected to the main network component
         scalableGeneratorConnector.connectGeneratorsTransformers(network, PRE_PROCESSING_COUNTRIES);
+        return incompleteShiftCountries;
+    }
+
+    private void checkGlskLimitation(Map<String, Double> incompleteShiftCountries, double mismatchEsPt, double mismatchEsFr) throws GlskLimitationException {
+        if (incompleteShiftCountries.containsKey(toEic("PT"))) {
+            checkGlskLimitationCountry("PT", incompleteShiftCountries.get(toEic("PT")), mismatchEsPt);
+        }
+        if (incompleteShiftCountries.containsKey(toEic("FR"))) {
+            checkGlskLimitationCountry("FR", incompleteShiftCountries.get(toEic("FR")), mismatchEsFr); //todo test
+        }
+        if (incompleteShiftCountries.containsKey(toEic("ES"))) {
+            checkGlskLimitationCountry("ES", incompleteShiftCountries.get(toEic("ES")), -(mismatchEsFr + mismatchEsPt));
+        }
+    }
 
-        if (!limitingCountries.isEmpty()) {
-            StringJoiner sj = new StringJoiner(", ", "There are Glsk limitation(s) in ", ".");
-            limitingCountries.forEach(sj::add);
-            LOGGER.error("[{}] : {}", direction, sj);
-            throw new GlskLimitationException(sj.toString());
+    private void checkGlskLimitationCountry(String country, double diffShift, double mismatch) throws GlskLimitationException {
+        // In case of asked > 0 : (done - asked) will be < 0, we have Glsk limitation if the next asked value increase (mismatch < 0),
+        // In case of asked < 0 : (done - asked) will be > 0, we have Glsk limitation if the next asked value decrease (mismatch > 0)
+        if (diffShift < 0 && mismatch < 0 || diffShift > 0 && mismatch > 0) {
```

Comment:
An easier way to check if both doubles have the same sign : 
`if(diffShift * mismatch > 0)`